File: solutions/_169_majority_element/solution.py
from typing import List


class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        count = 0
        maj = None

        for n in nums:
            if count == 0:
                maj = n
            if maj == n:
                count += 1
            else:
                count -= 1

        return maj


# https://leetcode.com/problems/majority-element/

File: solutions/_169_majority_element/solution.java
package _169_majority_element;

class Solution {

  public int majorityElement(int[] nums) {
    int count = 0;
    int maj = 0;

    for (int n : nums) {
      if (count == 0) {
        maj = n;
      }
      if (maj == n) {
        count++;
      } else {
        count--;
      }
    }

    return maj;
  }
}
/*
https://leetcode.com/problems/majority-element/
*/
